stackit/internal/services/objectstorage/objectstorage_acc_test.go

```diff
@@ -0,0 +1,156 @@
+		testutil.ObjectStorageProviderConfig(),
+		bucketResource["project_id"],
+		bucketResource["bucket_name"],
+	)
+}
+
+func TestAccObjectStorageResource(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testutil.TestAccProtoV6ProviderFactories,
+		CheckDestroy:             testAccCheckObjectStorageDestroy,
+		Steps: []resource.TestStep{
+
+			// Creation
+			{
+				Config: resourceConfig(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Instance data
+					resource.TestCheckResourceAttr("stackit_objectstorage_bucket.bucket", "project_id", bucketResource["project_id"]),
+					resource.TestCheckResourceAttr("stackit_objectstorage_bucket.bucket", "bucket_name", bucketResource["bucket_name"]),
+					resource.TestCheckResourceAttrSet("stackit_objectstorage_bucket.bucket", "url_path_style"),
+					resource.TestCheckResourceAttrSet("stackit_objectstorage_bucket.bucket", "url_virtual_hosted_style"),
+				),
+			},
+			// Data source
+			{
+				Config: fmt.Sprintf(`
+					%s
+
+					data "stackit_objectstorage_bucket" "bucket" {
+						project_id  = stackit_objectstorage_bucket.bucket.project_id
+						bucket_name = stackit_objectstorage_bucket.bucket.bucket_name
+					}`,
+					resourceConfig(),
+				),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Instance data
+					resource.TestCheckResourceAttr("data.stackit_objectstorage_bucket.bucket", "project_id", bucketResource["project_id"]),
+					resource.TestCheckResourceAttrPair(
+						"stackit_objectstorage_bucket.bucket", "bucket_name",
+						"data.stackit_objectstorage_bucket.bucket", "bucket_name",
+					),
+					resource.TestCheckResourceAttrPair(
+						"stackit_objectstorage_bucket.bucket", "url_path_style",
+						"data.stackit_objectstorage_bucket.bucket", "url_path_style",
+					),
+					resource.TestCheckResourceAttrPair(
+						"stackit_objectstorage_bucket.bucket", "url_virtual_hosted_style",
+						"data.stackit_objectstorage_bucket.bucket", "url_virtual_hosted_style",
+					),
+				),
+			},
+			// Import
+			{
+				ResourceName: "stackit_objectstorage_bucket.bucket",
+				ImportStateIdFunc: func(s *terraform.State) (string, error) {
+					r, ok := s.RootModule().Resources["stackit_objectstorage_bucket.bucket"]
+					if !ok {
+						return "", fmt.Errorf("couldn't find resource stackit_objectstorage_bucket.bucket")
+					}
+					bucketName, ok := r.Primary.Attributes["bucket_name"]
+					if !ok {
+						return "", fmt.Errorf("couldn't find attribute bucket_name")
+					}
+
+					return fmt.Sprintf("%s,%s", testutil.ProjectId, bucketName), nil
+				},
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			// Deletion is done by the framework implicitly
+		},
+	})
+}
+
+func testAccCheckObjectStorageDestroy(s *terraform.State) error {
+	ctx := context.Background()
+	var client *objectstorage.APIClient
+	var err error
+	if testutil.ObjectStorageCustomEndpoint == "" {
+		client, err = objectstorage.NewAPIClient()
```

Comment:
region config is missing here. If you copied this from other acc tests, it might be missing there too

Reply:
Ideally this should access the resource's API client, it shouldn't have to initialize a new one. But I couldn't find a way to do it. Settled for the quick fix of setting the region as "eu01", which is aligned with what `testutil.ObjectStorageProviderConfig()` (the provider config used for the acceptance tests) does